common/rulesys.cpp

```diff
@@ -24,462 +24,451 @@
+	LogRulesDetail("Loading Rule Set {} ({}).", rule_set_name, rule_set_id);
 
-	auto results = database->QueryDatabase(query);
-	if (!results.Success()) {
+	const auto& l = RuleValuesRepository::GetWhere(
+		*db,
+		fmt::format(
+			"ruleset_id = {}",
+			rule_set_id
+		)
+	);
+	if (l.empty()) {
 		return false;
 	}
 
-	for (auto row = results.begin(); row != results.end(); ++row) {
-		if (!SetRule(row[0], row[1], nullptr, false, reload)) {
-			Log(Logs::Detail, Logs::Rules, "Unable to interpret rule record for '%s'", row[0]);
+	for (const auto& e : l) {
+		if (!SetRule(e.rule_name, e.rule_value, nullptr, false, reload)) {
+			LogRulesDetail("Unable to interpret Rule record for Rule '{}'.", e.rule_name);
 		}
 	}
 
 	return true;
 }
 
-void RuleManager::SaveRules(Database *database, const char *ruleset_name) {
-
-	if (ruleset_name != nullptr) {
-		//saving to a specific name
-		if (m_activeName != ruleset_name) {
-			//a new name...
-
-			m_activeRuleset = _FindOrCreateRuleset(database, ruleset_name);
+void RuleManager::SaveRules(Database *db, std::string rule_set_name) {
+	if (!rule_set_name.empty()) {
+		if (m_activeName != rule_set_name) {
+			m_activeRuleset = _FindOrCreateRuleset(db, rule_set_name);
 			if (m_activeRuleset == -1) {
-				Log(Logs::Detail, Logs::Rules, "Unable to find or create rule set %s", ruleset_name);
+				LogRulesDetail("Unable to find or create Rule Set {}.", rule_set_name);
 				return;
 			}
-			m_activeName = ruleset_name;
+
+			m_activeName = rule_set_name;
 		}
-		Log(Logs::Detail, Logs::Rules, "Saving running rules into rule set %s (%d)", ruleset_name, m_activeRuleset);
-	}
-	else {
-		Log(Logs::Detail, Logs::Rules, "Saving running rules into running rule set %s", m_activeName.c_str(), m_activeRuleset);
+
+		LogRulesDetail("Saving running rules into Rule Set {} ({}).", rule_set_name, m_activeRuleset);
+	} else {
+		LogRulesDetail("Saving running rules into running Rule Set {} ({}).", m_activeName, m_activeRuleset);
 	}
 
 	int i;
 	for (i = 0; i < _IntRuleCount; i++) {
-		_SaveRule(database, IntRule, i);
+		_SaveRule(db, IntRule, i);
 	}
+
 	for (i = 0; i < _RealRuleCount; i++) {
-		_SaveRule(database, RealRule, i);
+		_SaveRule(db, RealRule, i);
 	}
+
 	for (i = 0; i < _BoolRuleCount; i++) {
-		_SaveRule(database, BoolRule, i);
+		_SaveRule(db, BoolRule, i);
 	}
 }
 
-void RuleManager::_SaveRule(Database *database, RuleType type, uint16 index) {
-	char value_string[100];
+void RuleManager::_SaveRule(Database *db, RuleType type, uint16 index) {
+	const auto rule_name = _GetRuleName(type, index);
 
-	if (type == IntRule && strcasecmp(_GetRuleName(type, index), "World:ExpansionSettings") == 0)
-		return;
-	if (type == BoolRule && strcasecmp(_GetRuleName(type, index), "World:UseClientBasedExpansionSettings") == 0)
+	if (
+		(type == BoolRule && Strings::Contains(rule_name, "World:UseClientBasedExpansionSettings")) ||
+		(type == IntRule && Strings::Contains(rule_name, "World:ExpansionSettings"))
```

Comment:
This also changes behavior `strcasecmp` is a comparison if strings are equal (case insensitive) - a contains is probably fine at least in this case

Reply:
Will change to `Strings::Equals`.